Source/Core/Core/IOS/DolphinDevice.cpp

```diff
@@ -123,6 +129,38 @@ IPCCommandResult SetSpeedLimit(const IOCtlVRequest& request)
   return DolphinDevice::GetDefaultReply(IPC_SUCCESS);
 }
 
+IPCCommandResult GetRealProductCode(const IOCtlVRequest& request)
+{
+  if (!request.HasNumberOfValidVectors(0, 1))
+  {
+    return DolphinDevice::GetDefaultReply(IPC_EINVAL);
+  }
+
+  const std::string backup_file_path = File::GetUserPath(D_BACKUP_IDX) + DIR_SEP + WII_SETTING;
+
+  File::IOFile file(backup_file_path, "rb");
+  if (!file)
+    return DolphinDevice::GetDefaultReply(IPC_ENOENT);
+
+  Common::SettingsHandler::Buffer data;
+
+  if (!file.ReadBytes(data.data(), data.size()))
+    return DolphinDevice::GetDefaultReply(IPC_ENOENT);
+
+  Common::SettingsHandler gen;
+  gen.SetBytes(std::move(data));
+  const std::string code = gen.GetValue("CODE");
+
+  const auto length = std::min(size_t(request.io_vectors[0].size), code.length());
```

Comment:
I would prefer:

```suggestion
  const size_t length = std::min<size_t>(request.io_vectors[0].size, code.length());
```

But it isn't too important.

Reply:
Done.